Python/Product/TestAdapter.Executor/PythonTestDiscoverer.cs

```diff
@@ -25,14 +25,39 @@
 
 namespace Microsoft.PythonTools.TestAdapter {
 
+    /// <summary>
+    /// Note even though we we specify  [DefaultExecutorUri(PythonConstants.UnitTestExecutorUriString)] we still get all .py source files
+    /// from all testcontainers.  
+    /// </summary>
+    [FileExtension(".py")]
+    [DefaultExecutorUri(PythonConstants.UnitTestExecutorUriString)]
+    public class UnitTestDiscoverer : PythonTestDiscoverer {
+        public UnitTestDiscoverer() : base(TestFrameworkType.UnitTest) {
+        }
+    }
 
     /// <summary>
-    /// Note even though we we specify  [DefaultExecutorUri(PythonConstants.PytestExecutorUri)] we still get all .py source files
+    /// Note even though we we specify  [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)] we still get all .py source files
     /// from all testcontainers.  
     /// </summary>
     [FileExtension(".py")]
     [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)]
+    public class PytestTestDiscoverer : PythonTestDiscoverer {
+        public PytestTestDiscoverer() : base(TestFrameworkType.Pytest) { 
+        }
+    }
+    
     public class PythonTestDiscoverer : ITestDiscoverer {
+        private TestFrameworkType _frameworkType;
+
+        /// <summary>
+        /// Create a test framework specific Test Discoverer
+        /// </summary>
+        /// <param name="frameworkType"></param>
+        public PythonTestDiscoverer(TestFrameworkType frameworkType) {
```

Comment:
not meant to be directly instantiated anymore, so you could make this ctor protected and mark the class abstract.